x-pack/filebeat/input/http_endpoint/config.go

```diff
@@ -110,6 +110,10 @@ func (c *config) Validate() error {
 		return errors.New("crc.provider is required when crc.secret is defined")
 	}
 
+	if c.URL == "" {
```

Comment:
The `URL` field can be marked with a struct tag.

```
URL                   string                  `config:"url" validate:"required"`
```

See https://pkg.go.dev/github.com/elastic/go-ucfg#Config.Unpack

Reply:
And continue to d ovalidation for `c.URL == ""` to return custom error message?

Reply:
It won't be necessary.